src/emulator/tasksEmulator.ts

```diff
@@ -0,0 +1,314 @@
+import * as express from "express";
+
+import { Constants } from "./constants";
+import { EmulatorInfo, EmulatorInstance, Emulators } from "./types";
+import { createDestroyer } from "../utils";
+import { EmulatorLogger } from "./emulatorLogger";
+import { TaskQueue } from "./taskQueue";
+import * as cors from "cors";
+
+export interface TasksEmulatorArgs {
+  port?: number;
+  host?: string;
+}
+
+export interface Task {
+  name: string;
+  // A timestamp in RFC3339 UTC "Zulu" format, with nanosecond resolution and up to nine fractional
+  // digits. Examples: "2014-10-02T15:01:23Z" and "2014-10-02T15:01:23.045123456Z".
+  scheduleTime?: string;
+  // A duration in seconds with up to nine fractional digits, terminated by 's'. Example: "3.5s".
+  dispatchDeadline?: string;
+  httpRequest: {
+    url: string;
+    oidcToken?: {
+      serviceAccountEmail: string;
+    };
+    body: any;
+    headers: { [key: string]: string };
+  };
+}
+
+export interface TaskQueueConfig {
+  retryConfig: RetryConfig;
+  rateLimits: RateLimits;
+  timeoutSeconds: number;
+  retry: boolean;
+  defaultUri: string;
+
+  // The configurations below this point do not currently have any effect on how the task queues are handled within the emulator
+  secrets?: string[]; // TODO(gburroughs): look into how we can handle this
+  region?: string | ResetValue;
+  memory?: string | ResetValue;
+  minInstances?: number | ResetValue;
+  maxInstances?: number | ResetValue;
+  concurrency?: number | ResetValue;
+  labels?: Record<string, string>;
+}
+
+export interface RetryConfig {
+  maxAttempts: number;
+  maxRetrySeconds: number | null;
+  maxBackoffSeconds: number;
+  maxDoublings: number;
+  minBackoffSeconds: number;
+}
+
+export interface RateLimits {
+  maxConcurrentDispatches: number;
+  maxDispatchesPerSecond: number;
+}
+
+type ResetValue = null;
```

Comment:
nit* ResetValue is typically associated with reconfiguring an existing resource deployed in cloud, so it won't necessarily an important consideration for the emulator where resources are ephemeral. 

(no action required)